packages/dataviews/src/view-list.tsx

```diff
@@ -40,7 +40,7 @@ interface ListViewProps {
 }
 
 interface ListViewItemProps {
-	id: string;
+	id: any;
```

Comment:
What's the proper type here?

Reply:
By reading the code it seems to be "string" but optional, so something like 
```suggestion
	id?: string;
```

Reply:
`string | undefined`?

Reply:
I'll add a bit of information, there's a subtle difference in what these two communicate:

```ts
// The property may not be present. If it is present it's a string
var x: { id?: string } = {};
var x2: { id?: string } = { id: undefined }; // TYPE ERROR!!
// The property _is_ present, `undefined` is an acceptable value. {} is not allowed here.
var y: { id: string | undefined } = { id: undefined };
var y2: { id: string | undefined } = {}; // TYPE ERROR!!
```

(Note that the `x2` type error depends on [`exactOptionalPropertyTypes`](https://www.typescriptlang.org/tsconfig/#exactOptionalPropertyTypes) which I don't _think_ we have enabled in Gutenberg right now. It's relatively new but likely a good thing to enable.)

At runtime we often treat these as the same, but they're also subtly different depending on the type of check we're doing:

```js
const x = {}
x.y; // undefined
Object.hasOwn(x, 'y'); // false
'y' in x; // false

x.y = undefined;
// The value is the same but the property is there:
x.y; // undefined
Object.hasOwn(x, 'y'); // true
'y' in x; // true
```